public/js/app/leaderboards.js

```diff
@@ -51,185 +39,130 @@ function leaderboardUpdate(data) {
-      } else {
-        backgroundColor = 'greyLeaderboard';
-      }
-      // Gets the player data and inserts it
-
-      let rating = data[playerIndex].value.rating;
-      let winRate = data[playerIndex].value.wonGames / data[playerIndex].value.totalgames * 100;
-      document.getElementById('onePlayer').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}' > ${data[playerIndex].label}</li>`);
-      document.getElementById('oneRank').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${-1 * (playerIndex - data.length + addNextPlayer)}</li>`);
-      document.getElementById('oneRating').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${rating.toFixed(0)}</li>`);
-      document.getElementById('oneGamesAmount').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${data[playerIndex].value.totalgames}</li>`);
-      document.getElementById('oneWon').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${winRate.toFixed(1)}% </li>`);
-
+      backgroundColor = 'greyLeaderboard';
     }
+    // Gets the player data and inserts it into the li element
+    let rating = playerList[playerIndex].value.rating;
+    let winRate = playerList[playerIndex].value.wonGames / playerList[playerIndex].value.totalgames * 100;
+    document.getElementById('onePlayer').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}' > ${playerList[playerIndex].label}</li>`);
+    document.getElementById('oneRank').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${-1 * (playerIndex - playerList.length)}</li>`);
+    document.getElementById('oneRating').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${rating.toFixed(0)}</li>`);
+    document.getElementById('oneGamesAmount').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${playerList[playerIndex].value.totalgames}</li>`);
+    document.getElementById('oneWon').insertAdjacentHTML('afterbegin', `<li class='leaderboardDelete ${backgroundColor}'> ${winRate.toFixed(1)}% </li>`);
   }
-  console.log(`playerIndex is ${playerIndex}`);
-  console.log(`addNextPlayer is ${addNextPlayer}`)
+}
+//This function triggers when the next, previous, first or last button are clicked
+let pageButton = document.querySelectorAll('.pageButton');
+function pageChange(newPageNumber) {
+  pageNumber = newPageNumber;
+  pageButton.forEach(element => element.classList.remove(`exhaustedButton`));
+  if (pageNumber === 0) {
+    //You see 4-7 pageButton because there are a total of 8 buttons counting the ones at the bottom of the page
+    pageButton[0].classList.add('exhaustedButton');
+    pageButton[3].classList.add('exhaustedButton');
+    pageButton[4].classList.add('exhaustedButton');
+    pageButton[7].classList.add('exhaustedButton');
```

Comment:
If someone modifies the HTML code or the order these buttons are displayed, your code breaks entirely without warning.
If these are different buttons, give them each a specific ID and disable them specifically
If these are not different buttons, make a for loop to group buttons and iterate over them to disable them in a single statement!